packages/jazz-tools/src/tests/coMap.test.ts

```diff
@@ -734,3 +734,74 @@ describe("CoMap applyDiff", async () => {
         expect((map as any).invalidField).toBeUndefined();
     });
 });
+
+describe("CoMap Typescript validation", async () => {
+    const me = await Account.create({
+        creationProps: { name: "Hermes Puggington" },
+        crypto: Crypto,
+    });
+
+    test("Should not validate if null is passed to a required ref", () => {
+        class TestMap extends CoMap {
+            required = co.ref(NestedMap);
+            optional = co.optional.ref(NestedMap);
+        }
+
+        class NestedMap extends CoMap {
+            value = co.string;
+        }
+
+        const map = TestMap.create(
+            {
+                optional: NestedMap.create({ value: "" }, { owner: me }),
+                // @ts-expect-error This should return error
+                required: null,
+            },
+            { owner: me },
+        );
+
+        // No runtime validation is implemented for now
+        expect(map).toBeTruthy();
```

Comment:
Wondering if we also should throw an error here.

From my experience falsy values on non-optional fields can lead to weird situations, so IMHO it would be better to throw early to avoid polluting the store.